src/apple/keychain.c

```diff
@@ -0,0 +1,236 @@
+
+#include <security/SecKeychain.h>
+#include <security/SecKey.h>
+#include <security/Security.h>
+
+#include "../keychain.h"
+#include "um_debug.h"
+
+static const char* keychain_strerror(OSStatus s) {
+    static char err[1024];
```

Comment:
This gonna bite us?

Reply:
it shouldn't but I updated to handle different scenarios after reading docs